### PR TITLE
[GH-25020] Add new resource links to [Help (?) drop-down menu] & icons

### PR DESCRIPTION
**Pull request by umrkhn**
#### Summary
This PR add new resource links to `Help (?) drop-down menu` also adds icons to existing links.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/25020
Jira https://mattermost.atlassian.net/browse/MM-55024

#### Screenshots

|  before  |  after  |
|----|----|
before | after |

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by umrkhn** [comment]
This PR is currently being worked on, will update it when its done.

**Comment by umrkhn** [comment]
Currently there is no `Give feedback` option, do I have to add it and where should it navigate to ?

**Comment by cwarnermm** [comment]
@sbishel @matthewbirtch - We'll need some Engineering and UX support for this PR, please.

**Comment by matthewbirtch** [comment]
> Currently there is no `Give feedback` option, do I have to add it and where should it navigate to ?

Let's keep the existing `Give feedback` menu option and it should behave exactly as it does right now (opening the feedbackbot). I believe this menu option will only display if you have the `User Satisfaction Surveys` plugin installed and enabled in the System Console

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Thanks for the contribution @umrkhn. A few minor UI details:

1. Can we adjust the icon color in the menu items to `color: rgba(var(--center-channel-color-rgb), 0.72);`
2. For some reason, the icons all look like they are sitting slightly too low. 
image

**Review by matthewbirtch — APPROVED** [review]
Looks good to me @umrkhn! Thanks for working on this one

**Review by cwarnermm — APPROVED** [review]
Thank you, @umrkhn!

File: webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/__snapshots__/user_guide_dropdown.test.tsx.snap [hunk 1]
```diff
@@ -49,25 +49,36 @@ exports[`components/channel_header/components/UserGuideDropdown should match sna
...
+        show={true}
+        text="Mattermost user guide"
+        url="https://docs.mattermost.com/guides/use-mattermost.html"
+      />
+      <MenuItemExternalLink
+        iconClassName="icon-lightbulb-outline"
+        id="trainingResourcesLink"
+        show={true}
+        text="Training resources"
+        url="https://academy.mattermost.com/courses"
```
Comment (cwarnermm) [inline webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/__snapshots__/user_guide_dropdown.test.tsx.snap:63]:
```suggestion
        url="https://academy.mattermost.com/"
```

File: webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/__snapshots__/user_guide_dropdown.test.tsx.snap [hunk 2]
```diff
@@ -127,18 +138,28 @@ exports[`components/channel_header/components/UserGuideDropdown should match sna
...
-        url="helpLink"
+        text="Mattermost user guide"
+        url="https://docs.mattermost.com/guides/use-mattermost.html"
       />
       <MenuItemExternalLink
+        iconClassName="icon-lightbulb-outline"
+        id="trainingResourcesLink"
+        show={true}
+        text="Training resources"
+        url="https://academy.mattermost.com/courses"
```
Comment (cwarnermm) [inline webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/__snapshots__/user_guide_dropdown.test.tsx.snap:152]:
```suggestion
        url="https://academy.mattermost.com/"
```

File: webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/__snapshots__/user_guide_dropdown.test.tsx.snap [hunk 3]
```diff
@@ -198,31 +219,43 @@ exports[`components/channel_header/components/UserGuideDropdown should match sna
...
+        show={true}
+        text="Mattermost user guide"
+        url="https://docs.mattermost.com/guides/use-mattermost.html"
+      />
+      <MenuItemExternalLink
+        iconClassName="icon-lightbulb-outline"
+        id="trainingResourcesLink"
+        show={true}
+        text="Training resources"
+        url="https://academy.mattermost.com/courses"
```
Comment (cwarnermm) [inline webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/__snapshots__/user_guide_dropdown.test.tsx.snap:233]:
```suggestion
        url="https://academy.mattermost.com/"
```

File: webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/user_guide_dropdown.tsx [hunk 4]
```diff
@@ -19,6 +19,8 @@ import {ModalIdentifiers} from 'utils/constants';
 
 import type {PropsFromRedux} from './index';
 
+const mattermostUserGuideLink = 'https://docs.mattermost.com/guides/use-mattermost.html';
+const trainingResourcesLink = 'https://academy.mattermost.com/courses';
```
Comment (cwarnermm) [inline webapp/channels/src/components/global_header/center_controls/user_guide_dropdown/user_guide_dropdown.tsx:23]:
```suggestion
const trainingResourcesLink = 'https://academy.mattermost.com/';
```

**Review by cwarnermm — APPROVED** [review]

**Review by crspeller — APPROVED** [review]

**Comment by cwarnermm** [comment]
@crspeller - Are you open to addressing the merge conflicts for this PR?

**Review by cwarnermm — APPROVED** [review]
Thank you, @umrkhn!!

**Comment by umrkhn** [comment]
@cwarnermm what's the status for this PR?

**Comment by cwarnermm** [comment]
@crspeller - Are you open to addressing the merge conflicts on this PR so that it can be merged?

**Pull request merged**